File: backend/internal/dto/article.go
package dto

import "time"

type CreateArticleRequest struct {
	Title       string `json:"title" binding:"required"`
	Content     string `json:"content" binding:"required"`
	Summary     string `json:"summary"`
	CoverImage  string `json:"cover_image"`
	IsPublished *bool  `json:"is_published"`
	Pinned      *bool  `json:"pinned"`
	CategoryID  uint   `json:"category_id"`
	TagIDs      []uint `json:"tag_ids"`
}

type UpdateArticleRequest struct {
	Title       string `json:"title"`
	Content     string `json:"content"`
	Summary     string `json:"summary"`
	CoverImage  string `json:"cover_image"`
	IsPublished *bool  `json:"is_published"`
	Pinned      *bool  `json:"pinned"`
	CategoryID  *uint  `json:"category_id"`
	TagIDs      []uint `json:"tag_ids"`
}

type ArticleListQuery struct {
	Page      int    `form:"page"`
	PageSize  int    `form:"page_size"`
	Category  string `form:"category"`
	Tag       string `form:"tag"`
	Keyword   string `form:"keyword"`
	Year      int    `form:"year"`
	Month     int    `form:"month"`
}

type ArticleResponse struct {
	ID          uint       `json:"id"`
	Title       string     `json:"title"`
	Slug        string     `json:"slug"`
	Content     string     `json:"content,omitempty"`
	Summary     string     `json:"summary"`
	CoverImage  string     `json:"cover_image"`
	IsPublished bool       `json:"is_published"`
	Pinned      bool       `json:"pinned"`
	ViewCount   int        `json:"view_count"`
	CategoryID  uint       `json:"category_id"`
	AuthorID    uint       `json:"author_id"`
	PublishedAt *time.Time `json:"published_at"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	Category    *CategoryInfo `json:"category,omitempty"`
	Author      *AuthorInfo   `json:"author,omitempty"`
	Tags        []TagInfo     `json:"tags,omitempty"`
}

type CategoryInfo struct {
	ID   uint   `json:"id"`
	Name string `json:"name"`
	Slug string `json:"slug"`
}

type AuthorInfo struct {
	ID       uint   `json:"id"`
	Username string `json:"username"`
	Nickname string `json:"nickname"`
	Avatar   string `json:"avatar"`
}

type TagInfo struct {
	ID   uint   `json:"id"`
	Name string `json:"name"`
	Slug string `json:"slug"`
}

type ArticleListResponse struct {
	ID          uint          `json:"id"`
	Title       string        `json:"title"`
	Slug        string        `json:"slug"`
	Summary     string        `json:"summary"`
	CoverImage  string        `json:"cover_image"`
	IsPublished bool          `json:"is_published"`
	Pinned      bool          `json:"pinned"`
	ViewCount   int           `json:"view_count"`
	CategoryID  uint          `json:"category_id"`
	PublishedAt *time.Time    `json:"published_at"`
	CreatedAt   time.Time     `json:"created_at"`
	Category    *CategoryInfo `json:"category,omitempty"`
	Tags        []TagInfo     `json:"tags,omitempty"`
}

File: frontend/src/types/article.ts
export interface Article {
  id: number;
  title: string;
  slug: string;
  content?: string;
  summary: string;
  cover_image: string;
  is_published: boolean;
  pinned: boolean;
  view_count: number;
  category_id: number;
  author_id: number;
  published_at: string | null;
  created_at: string;
  updated_at: string;
  category?: CategoryInfo;
  author?: AuthorInfo;
  tags?: TagInfo[];
}

export interface ArticleListItem {
  id: number;
  title: string;
  slug: string;
  summary: string;
  cover_image: string;
  is_published: boolean;
  pinned: boolean;
  view_count: number;
  category_id: number;
  published_at: string | null;
  created_at: string;
  category?: CategoryInfo;
  tags?: TagInfo[];
}

export interface CategoryInfo {
  id: number;
  name: string;
  slug: string;
}

export interface AuthorInfo {
  id: number;
  username: string;
  nickname: string;
  avatar: string;
}

export interface TagInfo {
  id: number;
  name: string;
  slug: string;
}

export interface CreateArticleRequest {
  title: string;
  content: string;
  summary?: string;
  cover_image?: string;
  is_published?: boolean;
  pinned?: boolean;
  category_id?: number;
  tag_ids?: number[];
}

export interface UpdateArticleRequest {
  title?: string;
  content?: string;
  summary?: string;
  cover_image?: string;
  is_published?: boolean;
  pinned?: boolean;
  category_id?: number;
  tag_ids?: number[];
}

export interface ArticleQuery {
  page?: number;
  page_size?: number;
  category?: string;
  tag?: string;
  keyword?: string;
  year?: number;
  month?: number;
}
